### PR TITLE
Update auth policy enum for proxy config

### PR DESCRIPTION
**Pull request by shamsher31**
Based on the https://github.com/istio/api/pull/1493#discussion_r447717857, the `ControlPlaneAuthPolicy` applied to Envoy > Istiod.

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -22,15 +22,15 @@ package istio.mesh.v1alpha1;
 
 option go_package="istio.io/api/mesh/v1alpha1";
 
-// AuthenticationPolicy defines authentication policy. It can be set for
+// AuthenticationPolicy defines authentication policy to be used by the proxy. It can be set for
```
Comment (smawson) [inline mesh/v1alpha1/proxy.proto:25]: Can we reword this to say:

AuthenticationPolicy defines how the proxy is authenticated when it connects to the control plane.

File: mesh/v1alpha1/proxy.proto [hunk 2]
```diff
@@ -22,15 +22,15 @@ package istio.mesh.v1alpha1;
...
 // different scopes (mesh, service …), and the most narrow scope with
```
Comment (smawson) [inline mesh/v1alpha1/proxy.proto:26]: I think this scope stuff is wrong? Where is it actually possible to set this? I only see it in ProxyConfig, so it can only be set at mesh level or at the level of an individual workload through an annotation? Or are there other ways to set ProxyConfig?

Sorry that's a lot of questions :)

File: mesh/v1alpha1/proxy.proto [hunk 3]
```diff
@@ -22,15 +22,15 @@ package istio.mesh.v1alpha1;
...
 // non-INHERIT value will be used.
 // Mesh policy cannot be INHERIT.
 enum AuthenticationPolicy {
-  // Do not encrypt Envoy to Envoy traffic.
+  // Do not encrypt Envoy to Istiod traffic.
```
Comment (smawson) [inline mesh/v1alpha1/proxy.proto:30]: Let's generalize to proxy to control plane traffic and not explicitly say envoy or istiod.

File: mesh/v1alpha1/proxy.proto [hunk 4]
```diff
@@ -309,7 +309,7 @@ message ProxyConfig {
   string availability_zone = 12 [deprecated=true];
 
   // Authentication policy defines the global switch to control authentication
```
Comment (smawson) [inline mesh/v1alpha1/proxy.proto:311]: Since this is in proxy config it isn't only global, we can reword this too.

File: mesh/v1alpha1/proxy.proto (hunk 3 above)
Comment (shamsher31) [reply mesh/v1alpha1/proxy.proto]: Sound good. Done.

File: mesh/v1alpha1/proxy.proto (hunk 1 above)
Comment (shamsher31) [reply mesh/v1alpha1/proxy.proto]: Perfect. Thanks.

File: mesh/v1alpha1/proxy.proto (hunk 2 above)
Comment (shamsher31) [reply mesh/v1alpha1/proxy.proto]: @smawson  As far as I know, it's only been used in ProxyConfig. In that case, I think the scope related sentence can be completely removed?

https://github.com/istio/api/blob/22f2acfe5ab4008a4d14009b3975752ed8dffc1b/mesh/v1alpha1/proxy.proto#L26-L28

Comment (smawson) [reply mesh/v1alpha1/proxy.proto]: AFAICT this can only be mesh-wide or set on a per-pod basis using the ProxyConfig annotation, so that would be two scopes: mesh or pod.

I'm OK removing the scope discussion or fixing it to be accurate.

Comment (shamsher31) [reply mesh/v1alpha1/proxy.proto]: Done, Thanks.

File: mesh/v1alpha1/proxy.proto [hunk 5]
```diff
@@ -22,15 +22,15 @@ package istio.mesh.v1alpha1;
 
 option go_package="istio.io/api/mesh/v1alpha1";
 
-// AuthenticationPolicy defines authentication policy. It can be set for
-// different scopes (mesh, service …), and the most narrow scope with
-// non-INHERIT value will be used.
+// AuthenticationPolicy defines how the proxy is authenticated when it connects to the control plane.
+// It can be set for two different scopes, mesh-wide or set on a per-pod basis using the ProxyConfig annotation.
+// and the most narrow scope with non-INHERIT value will be used.
```
Comment (smawson) [inline mesh/v1alpha1/proxy.proto:27]:
```suggestion
// The most narrow scope with non-INHERIT value will be used.
```

Or just delete this last line.

File: mesh/v1alpha1/proxy.proto [hunk 6]
```diff
@@ -22,15 +22,15 @@ package istio.mesh.v1alpha1;
 
 option go_package="istio.io/api/mesh/v1alpha1";
 
-// AuthenticationPolicy defines authentication policy. It can be set for
-// different scopes (mesh, service …), and the most narrow scope with
-// non-INHERIT value will be used.
+// AuthenticationPolicy defines how the proxy is authenticated when it connects to the control plane.
+// It can be set for two different scopes, mesh-wide or set on a per-pod basis using the ProxyConfig annotation.
+// The most narrow scope with non-INHERIT value will be used.
```
Comment (costinm) [inline mesh/v1alpha1/proxy.proto:27]: I'm not sure what inherit does - the ProxyConfig is either the default from MeshConfig, added via injection or the one user specifies in the injection. Nothing gets merged AFAIK. 

It may be worth adding few words about the default behavior: Istiod will automatically used MTLS if the destination port is 15012 or 443 ( I think I got both right ).

Comment (shamsher31) [reply mesh/v1alpha1/proxy.proto]: > I'm not sure what inherit does

Agree. I haven't seen the use of `INHERIT` anywhere in repo. Only `NONE` and `MUTUAL_TLS` is widely used across test cases and examples.
